db/migrate/20170106122400_link_documents_and_editions.rb

```diff
@@ -0,0 +1,10 @@
+class LinkDocumentsAndEditions < ActiveRecord::Migration[5.0]
+  def change
+    add_column :content_items, :document_id, :integer
+    add_foreign_key :content_items, :documents
+
+    add_index :content_items, [:document_id, :user_facing_version], unique: true
+    add_index :content_items, [:document_id, :content_store], unique: true
+    add_index :content_items, [:document_id, :state]
```

Comment:
These 3 will be locking as is, but can be changed to use the concurrent algorithm to not be so - though you have to disable the ddl transaction